documentation/directory-structure.md

```diff
@@ -0,0 +1,245 @@
+  - [Naming guidelines for `specification` folder contents](#naming-guidelines-for-specification-folder-contents)
+  - [`specification/common-types`](#specificationcommon-types)
+  - [Advanced scenario: service group](#advanced-scenario-service-group)
+    - [Service group `common-types`](#service-group-common-types)
+    - [Versioning services in a service group](#versioning-services-in-a-service-group)
+  - [Legacy layouts and OpenAPI specs](#legacy-layouts-and-openapi-specs)
+  - 
+# `specification` directory structure
+
+
+This article describes the directory structure / folder layout of the `specification` folder.
+You may be also interested in following:
+- [Specification index]
+- [Resource Provider list]
+
+> [!IMPORTANT]
+> The layout described in this article is strictly enforced. There exist some exceptions for historical reasons. These exceptions are not allowed going forward.
+
+## `specification` folder
+
+The `specification` folder is the root folder for all service specifications.
+
+Each child of the `specification` folder corresponds to a `service` specification for given Azure team. Here we denote such folder as `AzureTeamFolder`.
+
+Given `AzureTeamFolder` has following structure:
+
+- `AzureTeamFolder/TypeSpecFolder` (multiple folders)
+- `AzureTeamFolder/resource-manager`
+- `AzureTeamFolder/data-plane`
+
+The `AzureTeamFolder/TypeSpecFolder` folders contain the TypeSpec specification for the given `service`.
+You can find details on the name and contents of these folders in [TypeSpec directory structure].
+You can learn more about TypeSpec at [aka.ms/azsdk/typespec] and [aka.ms/typespec].
+
+## `resource-manager` and `data-plane` folders
+
+The `AzureTeamFolder/resource-manager` contains the ARM OpenAPI specifications emitted from TypeSpec in `AzureTeamFolder/TypeSpecFolder`.
+
+The `resource-manager` folder has exactly one child folder whose name matches the **Resource Provider (RP) Namespace** (`RPNS`), 
+such as `Microsoft.Automation` (full list of namespaces is [here][Resource Provider list]).
+There is 1 to 1 correspondence between an RP and an RP namespace.
+There must be **exactly one** RP namespace folder under given `resource-manager` folder.
+We denote such folder as `resource-manager/RPNS`.
+
+The `AzureTeamFolder/data-plane` contains the data-plane OpenAPI specifications emitted from TypeSpec in `TypeSpecFolder`. 
+The `data-plane` folder has no child `RPNS` folder. However, it can have a set of  `dataPlaneSubfolder` folders.
+
+## AutoRest configuration `README.md` files
+
+Both the `AzureTeamFolder/resource-manager` and `AzureTeamFolder/data-plane` folders must contain an AutoRest configuration file, `README.md`.
+Learn more about this file at [aka.ms/azsdk/autorest].
+
+<!-- This paragraph about README contents is a bit vague. We should improve it, eventually. -->
+
+Each `README.md` describes a single `service` and is used as an SDK package and documentation for each version of the service.
```

Comment:
We definitely have services that split into multiple SDK's so this is not necessarily 1:1. examples of KeyVault and HealthInsights come to mind.

Reply:
@weshaggard the question is - do we want to allow it going forward? If not, I can mention this in the 'Deprecated structure and hand-written OpenAPI specs' section

Reply:
@heaths might be able to explain more for KV. However, for HeathInsights I know the produce one spec which has multiple SDKs. See https://github.com/Azure/azure-rest-api-specs/tree/main/specification/ai/HealthInsights my understanding is this is allowed and not an exception.

Reply:
@weshaggard for HealthInsights you linked to TypeSpec folder. READMEs as in the data-plane subtree:
https://github.com/Azure/azure-rest-api-specs/tree/main/specification/ai/data-plane

There are multiple READMEs. But this case is already covered by the doc as merged in this PR, in the `service group` secrtion.